sql/parser/builtins.go

```diff
@@ -963,13 +980,13 @@ func datumToString(datum Datum) (string, error) {
 
 var replaceSubRe = regexp.MustCompile(`\\[&1-9]`)
 
-func regexpReplace(s, pattern, to, sqlFlags string) (Datum, error) {
+func regexpReplace(ctx EvalContext, s, pattern, to, sqlFlags string) (Datum, error) {
 	pattern, global, err := regexpEvalFlags(pattern, sqlFlags)
 	if err != nil {
 		return nil, err
 	}
 
-	patternRe, err := regexp.Compile(pattern)
+	patternRe, err := ctx.ReCache.GetRegexp(stringKey(pattern))
```

Comment:
I think you should avoid the call to `regexpEvalFlags` on cache hit by passing `regepKey{pattern, sqlFlags}`.